pvnet/models/multimodal/config_validation.py

```diff
@@ -0,0 +1,1007 @@
+    input_data_config: dict,
+    expected_batch_size: Optional[int] = None,
+) -> None:
+    """Validate a batch of numpy data against the multimodal configuration.
+
+    Performs static config validation first, then checks shapes of data arrays
+    in the batch against expectations derived from the config and input_data_config.
+
+    Args:
+        numpy_batch: Dictionary containing modality data as NumPy arrays.
+        multimodal_config: The multimodal configuration dictionary (for model settings).
+        input_data_config: The input data configuration dictionary (for data source settings).
+        expected_batch_size: Optional expected batch size to validate against.
+
+    Raises:
+        KeyError, TypeError, ValueError: If config is invalid or batch shapes mismatch config.
+    """
+
+    try:
+        _validate_static_config(multimodal_config)
+    except (KeyError, TypeError, ValueError) as e:
+        logger.error(f"Static configuration validation failed: {e}")
+        raise
+
+    logger.info(
+        f"Validating batch data shapes against configuration. Expected batch size: "
+        f"{expected_batch_size if expected_batch_size is not None else 'Not Provided'}."
+    )
+    cfg = multimodal_config
+    inferred_batch_size: int | None = None
+
+    # Satellite validation
+    if "sat_encoder" in cfg and cfg.get("sat_encoder"):
+        key = "satellite_actual"
+        modality_data_cfg_key = "satellite"
+        logger.debug(f"Validating modality: {key}")
+        if key not in numpy_batch:
+            raise KeyError(f"Batch missing '{key}' data, but 'sat_encoder' is configured.")
+        data = numpy_batch[key]
+        if not isinstance(data, np.ndarray):
+            raise TypeError(f"'{key}' data must be np.ndarray")
+
+        try:
+            interval = input_data_config[modality_data_cfg_key]['time_resolution_minutes']
+            if not isinstance(interval, int) or interval <= 0:
+                raise ValueError("Satellite interval must be a positive integer")
+        except KeyError:
+            msg = (
+                f"Could not find '{modality_data_cfg_key}.time_resolution_minutes' "
+                f"in input_data_config"
+            )
+            raise KeyError(msg)
+        except ValueError as e:
+            msg = f"Invalid satellite time_resolution_minutes in input_data_config: {e}"
+            raise ValueError(msg)
+
+        sat_cfg = _get_encoder_config(cfg, "sat_encoder", "sat_encoder")
+        h = w = sat_cfg["image_size_pixels"]
+        c = sat_cfg["in_channels"]
+        hist_mins = cfg.get("sat_history_minutes", cfg["history_minutes"])
+        hist_steps, _ = _get_time_steps(hist_mins, 0, interval)
+        expected_shape_no_batch = (hist_steps, c, h, w)
+
+        if data.ndim != 5:
+             raise ValueError(
+                f"'{key}' dimension error. Expected 5 dims (B, T, C, H, W), Got {data.ndim}"
+            )
+        if data.shape[1:] != expected_shape_no_batch:
```

Comment:
could use
`expected_shape_no_batch = (batch_size, hist_steps, c, h, w)`
and
`data.shape != expected_shape_no_batch`